src/classes/THAN_Filter_TDTM.cls

```diff
@@ -61,12 +61,47 @@ public class THAN_Filter_TDTM extends TDTM_Runnable {
 	                    String filterField = String.escapeSingleQuotes(handler.Filter_Field__c);
 	                    String obj = String.escapeSingleQuotes(handler.Object__c);
 	                    if(!String.isBlank(filterField) && !String.isBlank(obj)) {
-		                    try {
-		                        String dynamicQuery = 'select ' + filterField + ' from ' + obj + ' limit 1';
-		                        List<SObject> records =  Database.query(dynamicQuery);
-		                    } catch(QueryException e) {
-		                        handler.addError(Label.InvalidFilter + handler.Class__c);
-		                    }
+                            //those four type of fields are considered valid Filter field
+                            //1. Standard Field e.g. FirstName
+                            //2. HEDA Field WITHOUT namespace e.g. Military_Service__c
+                            //3. User's Custom Field e.g. MyField__c
+                            //4. Field from another package with namespace e.g. NS__Field__c
+
+                            String hedaPrefix = String.isBlank(UTIL_Namespace.getNamespace()) ? '' : (UTIL_Namespace.getNamespace() + '__');
+                            Boolean hasHEDAPrefix = filterField.startsWithIgnoreCase(hedaPrefix) ? true : false;
+                            Boolean ignoreDuplicateField = filterField.endsWithIgnoreCase(Label.duplicateFieldInFilter) ? true: false;
+
+                            //Get the ignoreDuplicateField flag and remove it from the Filter Field value
+                            if(ignoreDuplicateField) {
+                                filterField = filterField.replace(Label.duplicateFieldInFilter, '');
+                                handler.Filter_Field__c = handler.Filter_Field__c.replace(Label.duplicateFieldInFilter, '');
```

Comment:
Do we need line 77? Are we using handler.Filter_Field__c anywhere below?

Reply:
I think we need line 77 because this is the value that we are going to save to database. the filterField is just a variable in this method.

Reply:
Ah right, because we are running this on Before actions that run on Trigger_Handler__c. Never mind. :-)